ctapipe/tools/dl1_merge.py

```diff
@@ -0,0 +1,359 @@
+                    table_in.cols.cumulative_counts[row],
+                )
+
+        elif "/dl1/service" not in self.output_file:
+            target_group = self.output_file.create_group(
+                "/dl1", "service", createparents=True
+            )
+            for node in allowlist_service:
+                file.copy_node(node, newparent=target_group)
+
+    def merge_tables(self, file):
+        # Loop over all nodes listed in required_nodes. Appends table to output_file
+        # if it already exists, otherwise creates group and copies node. If skip_images
+        # or skip_simu_images or skip_parameters flag is True, related groups
+        # will be skipped.
+        for node in required_nodes:
+            if node in file:
+                if self.skip_images is True and node in blocklist_images:
+                    continue
+
+                if self.skip_simu_images is True and node in blocklist_images[0]:
+                    continue
+
+                if self.skip_parameters is True and node in blocklist_parameters:
+                    continue
+
+                if node in allowlist_service:
+                    continue
+
+                if node in allowlist_tels:
+                    if node not in self.output_file:
+                        head, tail = os.path.split(node)
+                        self.output_file.create_group(head, tail, createparents=True)
+                    for tel in file.root[node]:
+                        if (node + "/" + tel.name) in self.output_file:
+                            output_node = self.output_file.get_node(
+                                node + "/" + tel.name
+                            )
+                            input_node = file.root[node + "/" + tel.name]
+
+                            # cast needed for some image parameters that are sometimes
+                            # float32 and sometimes float64
+                            output_node.append(input_node[:].astype(output_node.dtype))
+                        else:
+                            target_group = self.output_file.root[node]
+                            file.copy_node(tel, newparent=target_group)
+                    continue
+                if node in self.output_file:
+                    # this is needed to merge 0.8 files due to
+                    # a column added erronouesly which prevents merging
+                    # because it's variable length
+                    # TODO: remove when we no longer want to support merging 0.8 files.
+                    data = file.root[node][:]
+                    if node == "/dl1/monitoring/subarray/pointing":
+                        data = self.drop_column(data, "tels_with_trigger")
+
+                    output_node = self.output_file.get_node(node)
+                    output_node.append(data)
+                else:
+                    group_path, table_name = os.path.split(node)
+                    if group_path not in self.output_file:
```

Comment:
This is the same code as above, right? So please move this to a small method like `get_or_create_group` and use it to avoid duplicatoin.